src/bokeh/models/renderers/glyph_renderer.py

```diff
@@ -216,7 +217,7 @@ def construct_color_bar(self, **kwargs: Any) -> ColorBar:
                 raise ValueError("expected text_color to be a field with a ColorMapper transform")
             return ColorBar(color_mapper=text_color.transform, **kwargs)
 
-        elif type(self.glyph) is Image:
+        elif type(self.glyph) in (Image, ImageStack):
```

Comment:
Note that using `type(self.glyph)` instead of `isinstance(self.glyph, ...)` prevents sub-classes to be allowed in this branch. If this is intentional, then I would suggest explaining the reason.

Reply:
It is intentional in the sense that this code was already using `type(self.glyph)` and hence I have kept it on the basis of making the minimal set of changes to existing code to achieve what I want.

I am happy for it to remain using `type` or change it to use `isinstance`. Would you prefer the latter?

Reply:
I generally prefer such trivial things to be fixed in the same PR, rather than spending another PR on this, or worse tripping over such issue at some point in the future. If it was a more elaborate fix, then sure another PR would be preferred.

Reply:
+1 for changing in this PR

Reply:
Switch to using `isinstance` in commit `e94dccd`.